### PR TITLE
Disassemble Title Screen

### PR DESCRIPTION
**Pull request by ehw**
I accidentally deleted the fork for the original pr so I'm recreating it with a new fork along with the changes I made since then.

### CODE REVIEW

File: engine/title.asm [hunk 1]
```diff
@@ -1,19 +1,648 @@
...
+	call VerticalMenu 			; 1D87
+	jp c, Init
+	ld a, [wMenuCursorY]				; CC2A
+	cp $01
+	jp z, Init
+
+	callfar InitAllSRAMBanks, $05
+	jp Init
+
+SRAMClear_TextMsg1::
```
Comment (xCrystal) [inline engine/title.asm]: formatting should be TAB + db + space + byte as in:

``
SRAMClear_TextChoice::
	db %11000000
``

File: engine/title.asm [hunk 2]
```diff
@@ -1,19 +1,648 @@
...
 endc
+	
 
-GameInit:: ; 62a5
+GameInit:: 								; 62A5
 	call ClearWindowData
 	ld a, $23
 	ld [wce5f], a
-	jp IntroSequence
+	jp IntroSequence
```
Comment (xCrystal) [inline engine/title.asm]: missing newline at end of file

File: engine/title.asm [hunk 3]
```diff
@@ -1,19 +1,648 @@
...
+FirePositionTable::					; 5EAC-5EB7 
+	dw $4CE0
+	dw $58A0
+	dw $6490
+	dw $70D0
+	dw $7CB0
+	dw $8800
+
+TitleFireGFX::	INCBIN "gfx/title/fire.2bpp"	; 5EB8-5F37 
+TitleNotesGFX::	INCBIN "gfx/title/notes.2bpp"	; 5F38=5FB7
```
Comment (xCrystal) [inline engine/title.asm]: tab before INCBIN should be space

**Comment by xCrystal** [comment]
Good job with this. Took the liberty to point out a few formatting flaws in case you missed them.

Related to this, do we need/want address comments on every function call/pointer? This is, I feel, redundant with the symfile, and I don't think we're doing it anywhere else (someone correct me if I'm wrong). In any case, the separation between the code and the comment should be a single space, and not a tab, and this includes address comments in the function header too, like ``IntroSequence:: ; 5D8C``.

Speaking of this, I think the convention is to use lowercase hex always (except as part of constant labels).

File: engine/title.asm (hunk 2 above)
Comment (ehw) [reply engine/title.asm]: Done.

File: engine/title.asm (hunk 3 above)
Comment (ehw) [reply engine/title.asm]: Done.

**Comment by ehw** [comment]
Agreed. I only left it in there as documentation for myself originally, but since those addresses are only good for Gold Debug and we allow Silver to be built I think it's not needed so I'll be removing it.

**Comment by ehw** [comment]
Made some more changes. Review?

**Comment by ehw** [comment]
I changed/updated from using Farcall/Callfar to callba/callab. I left the bank numbers as comments.

**Pull request merged**